pkg/azuredisk/nodeserver.go

```diff
@@ -538,10 +539,9 @@ func (d *Driver) findDiskAndLun(devicePath string) (string, int32, error) {
 		// wait until timeout
 		return false, nil
 	})
-	if err != nil {
-		return "", -1, err
+	if newDevicePath == "" {
```

Comment:
the logic is more clear now  @ZeroMagic

Reply:
```go
if newDevicePath, err = findDiskByLun(int(lun), io, d.mounter.Exec); err != nil {
			return false, fmt.Errorf("azureDisk - findDiskByLun(%v) failed with error(%s)", lun, err)
		}
```
If the error existed, the newDevicePath is still "", and the error will be changed here.

Reply:
good catch, fixed by:
```
	if err == nil && newDevicePath == "" {
		err = fmt.Errorf("azureDisk - findDiskByLun(%v) failed within timeout", lun)
	}
```